compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -541,46 +559,53 @@ trait ImplicitRunInfo { self: Run =>
         }
 
         val comps = new TermRefSet
-        tp match {
-          case tp: NamedType =>
-            if (!tp.symbol.is(Package) || ctx.scala2Mode) {
-              // Don't consider implicits in package prefixes unless under -language:Scala2
-              val pre = tp.prefix
-              comps ++= iscopeRefs(pre)
-              def addRef(companion: TermRef): Unit = {
-                val compSym = companion.symbol
-                if (compSym is Package) {
-                  assert(ctx.scala2Mode)
-                  addRef(companion.select(nme.PACKAGE))
-                }
-                else if (compSym.exists)
-                  comps += companion.asSeenFrom(pre, compSym.owner).asInstanceOf[TermRef]
-              }
-              def addCompanionOf(sym: Symbol) = {
-                val companion = sym.companionModule
-                if (companion.exists) addRef(companion.termRef)
+        def addCompanion(pre: Type, companion: Symbol) =
+          if (companion.exists && !companion.isAbsent) comps += TermRef(pre, companion)
```

Comment:
Use of `isAbsent` here breaks after https://github.com/lampepfl/dotty/pull/6842.

Reply:
What do you mean by break ? The signature of `isAbsent` changed, so at the very least you need to add parens to it: `isAbsent()`

Reply:
Well, yes, precisely :-)

Maybe my terminology is off here, but no git conflict, CI passes, yet will fail if merged.